doc/src/manual/parallel-computing.md

```diff
@@ -1,10 +1,265 @@
+
+We will consider first Julia  [Tasks (aka Coroutines)](@ref man-tasks) and other high-level interfaces and 
+modules that rely on the Julia runtime library, that allow to suspend and resume computations with full 
+control of inter-`Tasks` coommunication without worrying of the opeartive system's scheduler. Julia also
+allows to communicate between `Tasks` through operations like [`wait`](@ref) and [`fetch`](@ref).
+
+Julia supports also multi-threading , where execution is forked and a anonymous function is run across all
+threads. Described as a fork-join approach, all threads have to join together and serial execution continues.
+Multi-threading is supported using the `Base.Threads` module that is still considered experimental, as Julia is
+not fully thread-safe, consider for example [this issue](https://github.com/JuliaLang/julia/issues/22581).
+Also multi-threading should be implemented considering properly global variables, locks and atomics, so we will
+present it later.
+
+In the end, a proper mention will be made on distributed and parallel computing. With scientific computing 
+in mind, Julia natively implements interfaces to send a process through multiple cores or machines.
+Also we will mention useful external packages for distributed programming like `MPI.jl` and `DistributeArrays.jl`.
+
+
+## Scheduling
+
+Julia's parallel programming platform uses [Tasks (aka Coroutines)](@ref man-tasks) to switch among multiple
+computations. Whenever code performs a communication operation like [`fetch`](@ref) or [`wait`](@ref),
+the current task is suspended and a scheduler picks another task to run. A task is restarted when
+the event it is waiting for completes.
+
+For many problems, it is not necessary to think about tasks directly. However, they can be used
+to wait for multiple events at the same time, which provides for *dynamic scheduling*. In dynamic
+scheduling, a program decides what to compute or where to compute it based on when other jobs
+finish. This is needed for unpredictable or unbalanced workloads, where we want to assign more
+work to processes only when they finish their current tasks.
+
+As an example, consider computing the singular values of matrices of different sizes:
+
+```julia-repl
+julia> M = Matrix{Float64}[rand(800,800), rand(600,600), rand(800,800), rand(600,600)];
+
+julia> pmap(svd, M);
+```
+
+If one process handles both 800×800 matrices and another handles both 600×600 matrices, we will
+not get as much scalability as we could. The solution is to make a local task to "feed" work to
+each process when it completes its current task. For example, consider a simple [`pmap`](@ref)
+implementation:
+
+```julia
+function pmap(f, lst)
+    np = nprocs()  # determine the number of processes available
+    n = length(lst)
+    results = Vector{Any}(n)
+    i = 1
+    # function to produce the next work item from the queue.
+    # in this case it's just an index.
+    nextidx() = (global i; idx=i; i+=1; idx)
```

Comment:
There is no need for the `global i` or?

Reply:
Retried quickly, without the global index in function declaration the method returns "UndefVarErr : i not defined" ecc. Yeah globals are not very efficient, but I kept this example as a quick scratch.

Reply:
The `global` shouldn't be necessary, but IMHO this whole example should be rewritten more clearly to something like:

```julia
julia> function my_pmap(f, list)
           queue = collect(indices(list))
           results = Vector{Any}(length(list))
           @sync for worker in workers()
               @async while length(queue) > 0
                   index = pop!(queue)
                   results[index] = remotecall_fetch(f, worker, list[index])
               end
           end
           results
       end
my_pmap (generic function with 1 method)

julia> my_pmap(x->x+1, 5:10)
6-element Array{Any,1}:
  6
  7
  8
  9
 10
 11
```